src/cluster_legacy.c

```diff
@@ -3595,6 +3709,21 @@ void clusterBroadcastMessage(clusterMsgSendBlock *msgblock) {
 /* Build the message header. hdr must point to a buffer at least
  * sizeof(clusterMsg) in bytes. */
 static void clusterBuildMessageHdr(clusterMsg *hdr, int type, size_t msglen) {
```

Comment:
I would suggest just creating two header building functions: `clusterBuildMessageHdr` and `clusterBuildLightMessageHdr`. It reads better and avoids casting. The duplication is pretty trivial IMO.

Reply:
I would keep it the same, as we would need to add a condition in `createClusterMsgSendBlock` while calling hdr build

Reply:
>as we would need to add a condition in createClusterMsgSendBlock while calling hdr build

which condition is this? casting is dangerous and I think we can easily remove it by duplicating a little bit of code.

Reply:
Done